src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/BrowserInteropHelper.cs

```diff
@@ -154,93 +115,10 @@ internal static UnsafeNativeMethods.IServiceProvider HostHtmlDocumentServiceProv
-
-                // See HostHtmlDocumentServiceProvider property get accessor for more information on the use of
-                // this field to ensure we got a valid service provider.
-                _initializedHostScript.Value = true;
-            }
-        }
-
-        private static void HostFilterInput(ref MSG msg, ref bool handled)
-        {
-            WindowMessage message = (WindowMessage)msg.message;
-            // The host gets to see input, keyboard and mouse messages.
-            if (message == WindowMessage.WM_INPUT ||
-                (message >= WindowMessage.WM_KEYFIRST && message <= WindowMessage.WM_IME_KEYLAST) ||
-                (message >= WindowMessage.WM_MOUSEFIRST && message <= WindowMessage.WM_MOUSELAST))
-            {
-                if (MS.Win32.NativeMethods.S_OK == ForwardTranslateAccelerator(ref msg, false))
-                {
-                    handled = true;
-                }
-            }
-        }
-
-        /// <summary> This hook gets a "last chance" to handle a key. Such applicaton-unhandled
-        /// keys are forwarded to the browser frame.
-        /// </summary>
-        internal static IntPtr PostFilterInput(IntPtr hwnd, int msg, IntPtr wParam, IntPtr lParam, ref bool handled)
-        {
-            if (!handled)
-            {
-                if ((WindowMessage)msg >= WindowMessage.WM_KEYFIRST && (WindowMessage)msg <= WindowMessage.WM_IME_KEYLAST)
-                {
-                    MSG m = new MSG(hwnd, msg, wParam, lParam, SafeNativeMethods.GetMessageTime(), 0, 0);
-                    if (MS.Win32.NativeMethods.S_OK == ForwardTranslateAccelerator(ref m, true))
-                    {
-                        handled = true;
-                    }
-                }
-            }
-            return IntPtr.Zero;
-        }
-
-        internal static void InitializeHostFilterInput()
-        {
-            ComponentDispatcher.ThreadFilterMessage +=
-                                new ThreadMessageEventHandler(HostFilterInput);
-        }
-
-        private static void EnsureScriptInteropAllowed()
-        {
-            if (_isScriptInteropDisabled.Value == null)
-            {
-                _isScriptInteropDisabled.Value = SafeSecurityHelper.IsFeatureDisabled(SafeSecurityHelper.KeyToRead.ScriptInteropDisable);
-            }
-        }
-
-        private static SecurityCriticalDataForSet<HostingFlags> _hostingFlags;
         private static SecurityCriticalDataForSet<bool> _isInitialViewerNavigation;
-        private static SecurityCriticalDataForSet<bool?> _isScriptInteropDisabled;
         
         private static SecurityCriticalDataForSet<UnsafeNativeMethods.IServiceProvider> _hostHtmlDocumentServiceProvider;
         private static SecurityCriticalDataForSet<bool> _initializedHostScript;
-
-        [DllImport(ExternDll.PresentationHostDll, EntryPoint="ForwardTranslateAccelerator")]
```

Comment:
I believe you can go even further and delete PresentationHostDll and all its references and also delete anything that would require the file PresentationHost_cor3.dll to be used.

AFAIK PresentationHost_cor3.dll never shipped with any version of .Net Core (I checked in every version sine 3.0 and I didn't see it in any runtime packs).

Reply:
Yeah, I was about to. It never shipped indeed. Thanks for checking as well, Thomas.

Reply:
Fun fact, while digging into the PresentationHost stuff as whole, I've found out most of the `SecureEnvironment` (and AD RMS) cannot actually work and will throw (I believe `Msdrm` is not really supported anymore anyways on Windows newer than 8), either for missing dll (`PresentationHost_cor3`) or you get `STATUS_STACK_BUFFER_OVERRUN` somewhere in `secproc.dll` while calling the methods.

The DRM stuff is something I've never done (and most I'd say) but yeah, even on `NetFx` I had to manually load `PresentationHost_v0400` via `LoadLibrary` to at least retrieve `E_DRM_SERVICE_NOT_FOUND`, it won't load it on its own. At least it doesn't crash though after manually loading the library though!

There's literally zero documentation to it, kind of a black hole. Peak documentation is "TBD".

Reply:
Anyways, resolving the conversation, I'll address this in a separate PR, almost done with it.